examples/tutorial.dx

```diff
@@ -0,0 +1,483 @@
+'# Dex Tutorial
+
+' Dex is a functional, statically typed language for array processing. There are
+  many tools for array processing, from high-level libraries like NumPy and
+  MATLAB to low-level languages like CUDA. Dex gives you many of the safety and
+  simplicity benefits of high-level array processing languages, without
+  requiring that users give up low-level control.
+
+
+include "plot.dx"
+
+
+'## Array comprehensions
+
+' Before getting into language details, let us begin with the most useful
+  component of Dex, the `for` builder. The best analogy for this construct is
+  list comprehensions in Python. For instance, in Python, we might write a
+  list comprehension like:
+
+' `x = [[1 for j in range(10)] for i in range(5)]`
+
+' In Dex, this construct would be written as:
+
+x = for i:(Fin 5). for j:(Fin 10). 1
+
+' Once we have an variable, we can print it `:p`
+
+:p x
+
+' More interestingly, we can also see its type with `:t`. This type tells us
+  that `x` is a two-dimensional array, whose first dimension has size 5 and
+  second dimension has size 10.
+
+:t x
+
+' Once we have an array, we can use it in new comprehensions. For example,
+  let's try to add `5` to each array element. In Python, one might write this as:
+
+' `y = [[x[i][j] for j in range(10)] for i in range(5)]`
+
+' Dex can do something similar. The main superficial difference is the
+  array indexing syntax, which uses `array.i` instead of square brackets for
+  subscripting.
+
+y = for i:(Fin 5). for j:(Fin 10). x.i.j + 5
+
+:p y
+
+' However, we can make this expression nicer. Because `x` has a known array type
+  and `i` and `j` index into that type, Dex can infer the range of the loop.
+  That means that we can safely remove the explicit `Fin` type annotations and
+  get the same result.
+
+y' = for i. for j. x.i.j + 5
+
+' We can further reduce this array by applying array reduction functions like
+  `sum`:
+
+z = for i. sum x.i
+
+' This style of using `for` to construct type-inferred arrays is central to what
+  makes Dex powerful. Let's consider another example. This one produces a list of
+  length 50 in Python.
+
+' `x = [1 for j in range(10) for i in range(5)]`
+
+' The analogous array construct in Dex is written in the following form. It
+  produces a one-dimensional array of 50 elements.
+
+x2 = for (i, j): (Fin 5 & Fin 10). 1
```

Comment:
It might be worth mentioning that `&` is a tuple type before this line, since otherwise people might think that it's just some kind of syntax sugar (but it isn't!).